assets/eip-5058/ERC5058.sol

```diff
@@ -0,0 +1,273 @@
+
+/**
+ * @dev Implementation ERC721 Lockable Token
+ */
+abstract contract ERC5058 is ERC721, IERC5058 {
+    // Mapping from token ID to unlock time
+    mapping(uint256 => uint256) public lockedTokens;
+
+    // Mapping from token ID to lock approved address
+    mapping(uint256 => address) private _lockApprovals;
+
+    // Mapping from owner to lock operator approvals
+    mapping(address => mapping(address => bool)) private _lockOperatorApprovals;
+
+    /**
+     * @dev See {IERC5058-lockApprove}.
+     */
+    function lockApprove(address to, uint256 tokenId) public virtual override {
+        require(!isLocked(tokenId), "ERC5058: token is locked");
+        address owner = ERC721.ownerOf(tokenId);
+        require(to != owner, "ERC5058: lock approval to current owner");
+
+        require(
+            _msgSender() == owner || isLockApprovedForAll(owner, _msgSender()),
+            "ERC5058: lock approve caller is not owner nor approved for all"
+        );
+
+        _lockApprove(owner, to, tokenId);
+    }
+
+    /**
+     * @dev See {IERC5058-getLockApproved}.
+     */
+    function getLockApproved(uint256 tokenId) public view virtual override returns (address) {
+        require(_exists(tokenId), "ERC5058: lock approved query for nonexistent token");
+
+        return _lockApprovals[tokenId];
+    }
+
+    /**
+     * @dev See {IERC5058-lockerOf}.
+     */
+    function lockerOf(uint256 tokenId) public view virtual override returns (address) {
+        require(_exists(tokenId), "ERC5058: locker query for nonexistent token");
+        require(isLocked(tokenId), "ERC5058: locker query for non-locked token");
+
+        return _lockApprovals[tokenId];
+    }
+
+    /**
+     * @dev See {IERC5058-setLockApprovalForAll}.
+     */
+    function setLockApprovalForAll(address operator, bool approved) public virtual override {
+        _setLockApprovalForAll(_msgSender(), operator, approved);
+    }
+
+    /**
+     * @dev See {IERC5058-isLockApprovedForAll}.
+     */
+    function isLockApprovedForAll(address owner, address operator) public view virtual override returns (bool) {
+        return _lockOperatorApprovals[owner][operator];
+    }
+
+    /**
+     * @dev See {IERC5058-isLocked}.
+     */
+    function isLocked(uint256 tokenId) public view virtual override returns (bool) {
+        return lockedTokens[tokenId] > block.number;
+    }
+
+    /**
+     * @dev See {IERC5058-lockFrom}.
+     */
+    function lockFrom(
+        address from,
+        uint256 tokenId,
+        uint256 expired
+    ) public virtual override {
+        //solhint-disable-next-line max-line-length
+        require(_isLockApprovedOrOwner(_msgSender(), tokenId), "ERC5058: lock caller is not owner nor approved");
+        require(expired > block.number, "ERC5058: expired time must be greater than current block number");
+        require(!isLocked(tokenId), "ERC5058: token is locked");
+
+        _lock(_msgSender(), from, tokenId, expired);
```

Comment:
What do you mean by `from` here? msg.sender? token owner?
It seems like `from` is the token owner, so what about naming it as `owner` instead of `from`?

Reply:
Ok, I have replaced `from` with `owner`.

Reply:
Thanks a lot! You can resolve this one too. ;)